mgc/discrim/base.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+from sklearn.utils import check_X_y
+import numpy as np
+from .._utils import euclidean
+
+
+class DiscriminabilityTest(ABC):
+    r"""
+    A base class for a Discriminability test.
+    """
+
+    def __init__(self):
+        self.pvalue_ = None
+        super().__init__()
+
+    # @abstractmethod
+    def _statistic(self, x, y):
+        r"""
+        Calulates the independence test statistic.
+            
+        Parameters
+        ----------
+        x, y : ndarray
+            Input data matrices.
+        """
+
+        rdfs = self._discr_rdf(x, y)
+        stat = np.nanmean(rdfs)
+
+        return stat
+
+    def _discr_rdf(self, dissimilarities, labels):
+        # calculates test statistics distribution
+        rdfs = []
+
+        for i, label in enumerate(labels):
+            di = dissimilarities[i]
+
+            # All other samples except its own label
+            idx = labels == label
+            Dij = di[~idx]
+
+            # All samples except itself
+            idx[i] = False
+            Dii = di[idx]
+
+            rdf = [
+                1 - ((Dij < d).sum() + 0.5 * (Dij == d).sum()) / Dij.size for d in Dii
+            ]
+            rdfs.append(rdf)
+
+        out = np.full((len(rdfs), max(map(len, rdfs))), np.nan)
+        for i, rdf in enumerate(rdfs):
+            out[i, : len(rdf)] = rdf
+
+        return out
+
+    def _perm_stat(self, index):
```

Comment:
don't need this if it's empty and not an abstract method

Reply:
https://github.com/neurodata/mgc/blob/8950dbc95e34818b226e177a9331fa37b7fbbaaf/mgc/discrim/base.py#L57-L72